openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -644,6 +636,15 @@ OpenAssessment.ResponseView.prototype = {
             }
         }
 
+        if (this.getSavedFileCount() + files.length > this.data.MAXIMUM_FILE_UPLOAD_COUNT ) {
```

Comment:
`$('.submission__answer__file__block').length;` gets you the count of all files, deleted and not, which is useful for keeping track of indices, but I think you need to make this `$('.submission__answer__file__block').filter(':parent').length` for checking MAX_FILES

Reply:
How is it different?

Reply:
yeah, we're leaving an empty div for deleted files. `:parent` only includes non-empty elements

Reply:
done